src/core/connection.c

```diff
@@ -5940,6 +5941,16 @@ QuicConnResetIdleTimeout(
         IdleTimeoutMs = Connection->Settings.HandshakeIdleTimeoutMs;
     }
 
+    if (Path->EcnValidationState == ECN_VALIDATION_TESTING) {
+        uint64_t Now = CxPlatTimeUs64();
+        if (CxPlatTimeAtOrBefore64(Now, Path->EcnTestingEndingTime)) {
+            //
+            // Give ECN validation enough time to test ECN capability on the path.
+            //
+            IdleTimeoutMs += US_TO_MS(CxPlatTimeDiff64(Now, Path->EcnTestingEndingTime));
```

Comment:
Why would you extend the idle time just to keep testing the network? If they want to kill the connection sooner, we should let them, right?

Reply:
Because we can't tell if the connection is idle or all packets sent in ECN testing state are dropped. We have a similar idle timeout extending logic right below this block.

Reply:
If we're actively sending packets, then we aren't idle. I don't think this change should be necessary.

Reply:
removed